source/common/http/header_map_impl.cc

```diff
@@ -150,6 +150,46 @@ void HeaderString::append(const char* data, uint32_t size) {
   string_length_ += size;
 }
 
+void HeaderString::rtrim() {
+  switch (type_) {
+  case Type::Reference: {
```

Comment:
Can you tell me more about why you needed to implement rtrim for Reference types?

Reply:
That's a good question. I see test failures since tackling only `Inline` is not enough. I don't know for sure if `Reference` is being used. 
It seems not hard to provide rtrim for Reference so I blindly add the functionality. 

Do you have any context(or any guideline) of Reference usage?

Reply:
I see your snippets in 1.12. Moving conversation there